OpenRA.Mods.Common/Traits/Player/PlaceBuilding.cs

```diff
@@ -110,7 +110,7 @@ static void PlayBuildAnim(Actor self, ActorInfo unit)
 					&& x.Actor.Info.Traits.Get<ProductionInfo>().Produces.Intersect(bi.Queue).Any())
 					.ToList();
 			var producer = producers.Where(x => x.Actor.IsPrimaryBuilding()).Concat(producers)
-				.FirstOrDefault();
+				.First();
```

Comment:
Maybe keep this as FirstOrDefault, and then add a null check and early return?  We'll want this to be bulletproof against things dying in the same tick.